packages/types/src/index.ts

```diff
@@ -1,11 +1,32 @@
 /*
  * Reusable shapes for argument values
  */
+
+export enum DialogElementType {
+  Text = 'text',
+  TextArea = 'textarea',
+  Select = 'select',
+}
+
+export enum DialogElementSubType {
+  Email = 'email',
+  Number = 'number',
+  Telephone = 'tel',
+  Url = 'url',
+}
+
+export enum DataSource {
+  Users = 'users',
+  Channels = 'channels',
+  Conversations = 'conversations',
+  External = 'external',
+}
+
 export interface Dialog {
   title: string;
   callback_id: string;
   elements: {
-    type: 'text' | 'textarea' | 'select';
```

Comment:
I'm afraid that changing union type to enum may be fully backward-compatible. I cannot tell what the possible troubles are by this but at least, unions and enums are not the same. To me, just naming the union types would be a safer refactoring. What do you think?

Reply:
Just my two cents, but I really love enums more than unions aha